### DISCUSSION TITLE
Using TS , Not `children` filed in props

### DISCUSSION THREAD

**Discussion by gd-cho** [vercel/next.js#35995] [Help] [answered]
When I use  `npx create-next-app@latest --typescript` to build Next.js For TS , I'm import type NextPage has no `children` filed in props.
**The props was a empty object.**

![image](https://user-images.githubusercontent.com/75920170/162409015-274f8e00-8ea5-4b38-ac21-09343f316dba.png)

_dep version:_
**"next": "12.1.4",
"react": "18.0.0",
"react-dom": "18.0.0"**

**Comment by icyJoseph** [comment, marked as answer]
Why would a page need to have that prop type built in?

Are you trying to do something like this?

```tsx
// pages/another-page.tsx
import MyPage from './pages/my-page'

const AnotherPage = ()=> <MyPage><div>Injecting content</div></MyPage>

export default AnotherPage;
```

While that's technically possible, it seems prone to grow in complexity.

A `NextPage` conceptually is a top level abstraction that represents an entire route/page, and it's not necessarily a React Component. 

Why not remove the `NextPage` type, and move that to be a `Layout` Component, living for example in `components/`, and then use it as such, a Layout component, across pages.

The docs document a case that resembles my assumption, https://nextjs.org/docs/basic-features/layouts#with-typescript.

**Reply by gd-cho** [reply]
I want to use children filed. How can I use?
```javascript
const Layout: NextPage = (props) => {
  return (
    <>
      <main>{props.children}</main>
    </>
  );
};
```

**Reply by icyJoseph** [reply]
In this case, `NextPage` is not the type you want. Either do:

```tsx
import type {ReactNode} from 'react';

type LayoutProps = {children?: ReactNode}
const Layout = (props: LayoutProps) => <div>{props.children}</div>
```

Or just use `FC`

```tsx
import type {FC} from 'react';

const Layout: FC = (props) => <div>{props.children}</div>
```

Here's a bit more to read: https://www.carlrippon.com/react-children-with-typescript/ - but the learning is that `NextPage` is not a type to compose. It is something that Next uses as handover type between React components and their routing system.

In practice a NextPage cannot really have children coming through props. NextPage's can only have serialisable data coming in as props.

**Reply by gd-cho** [reply]
"@types/react": "18.0.0"  have not the children filed , same as well as use FC type.

**Reply by icyJoseph** [reply]
Oh nice, I didn't know they removed the implied children from the react 18 types, that's actually nice!

Just type if yourself then, using `ReactNode`.

```ts
type LayoutProps = {children?: ReactNode}
```

**Reply by azizoid** [reply]
Still don't get why they did that, have to type it all the time manually
